src/Bicep.Core/TypeSystem/TypeCollapser.cs

```diff
@@ -445,25 +447,31 @@ private static (TypeSymbol? type, TypePropertyFlags flags) GetAdditionalProperti
                     {
                         possibleTypes.Add(addlPropertiesType);
                     }
+
                     propertyFlags &= @object.AdditionalPropertiesFlags;
+                    
+                    if (@object.AdditionalPropertiesDescription is not null)
+                    {
+                        propertyDescription = @object.AdditionalPropertiesDescription;
+                    }
```

Comment:
I think the if block is unnecessary -- since `propertyDescription` is a nullable string, wouldn't an unconditional assignment do the same thing?

Reply:
I thought it's needed for object collapsing because of a case like the one used in [this test](https://github.com/Azure/bicep/blob/2629b5bc868a8f6f611254c7ff37f85f9c69ecb5/src/Bicep.Core.UnitTests/TypeSystem/TypeHelperTests.cs#L377-L412); without the check, the additional properties description is set to `null` by the `foo` type.